internal/engine/portforward/subscriber_test.go

```diff
@@ -467,30 +414,113 @@ func (f pfsFixture) WaitUntilDone() {
 	}
 }
 
-func (f *pfsFixture) forwardMatches(fwd Forward, expectedLocal, expectedContainer int32) bool {
-	return expectedLocal == fwd.LocalPort && expectedContainer == fwd.ContainerPort
+func (f *pfsFixture) forwardMatches(fwd Forward, expected expectedPF) bool {
+	if expected.local != 0 && expected.local != fwd.LocalPort {
+		return false
+	}
+	if expected.container != 0 && expected.container != fwd.ContainerPort {
+		return false
+	}
+	if expected.host != "" && expected.host != fwd.Host {
+		return false
+	}
+	return true
 }
 
-func (f *pfsFixture) oneForwardMatches(expectedLocal, expectedContainer int32, pf *PortForward) bool {
-	return len(pf.Spec.Forwards) == 1 && f.forwardMatches(pf.Spec.Forwards[0], expectedLocal, expectedContainer)
+func (f *pfsFixture) oneForwardMatches(pf *PortForward, expected expectedPF) bool {
+	if len(pf.Spec.Forwards) != 1 {
+		return false
+	}
+
+	if !f.forwardMatches(pf.Spec.Forwards[0], expected) {
+		return false
+	}
+	return expected.mName == "" || pf.ObjectMeta.Annotations[v1alpha1.AnnotationManifest] == expected.mName
 }
 
-func (f *pfsFixture) forwardWithHostMatches(fwd Forward, expectedLocal, expectedContainer int32, expectedHost string) bool {
-	return expectedLocal == fwd.LocalPort &&
-		expectedContainer == fwd.ContainerPort &&
-		expectedHost == fwd.Host
+// Use this func to assert on an expected PortForward with a single Fwd.
+// For more complicated assertions, use waitUntilStateAndAPIPortForward directly.
+func (f *pfsFixture) waitUntilStateAndAPIPortForwardAsExpected(msg string, expected expectedPF) {
+	f.T().Helper()
+
+	if expected.podID == "" {
+		f.T().Fatal("must pass pod ID as part of expectedPF")
+	}
+
+	f.waitUntilStateAndAPIPortForward(msg, pfName(expected.podID), func(pf *PortForward) bool {
+		return f.oneForwardMatches(pf, expected)
+	})
 }
 
-func (f *pfsFixture) oneForwardWithHostMatches(expectedLocal, expectedContainer int32, expectedHost string, pf *PortForward) bool {
-	return len(pf.Spec.Forwards) == 1 && f.forwardWithHostMatches(pf.Spec.Forwards[0], expectedLocal, expectedContainer, expectedHost)
+func (f *pfsFixture) waitUntilStateAndAPIPortForward(msg string, name string, pfOK func(pf *PortForward) bool) {
+	f.T().Helper()
+	f.waitUntilStatePortForwards(msg, func(pfs map[string]*PortForward) bool {
+		if len(pfs) != 1 {
+			return false
+		}
+
+		var pf *PortForward
+		for _, portForward := range pfs { // there's only one thing in the map, we just don't know its key 🙃
```

Comment:
Isn't its key the same as the API key below?

Reply:
...maybe 😓